### PR TITLE
Move all `libdnf5::Base` members to pImpl and remove deprecated

### PR DESCRIPTION
**Pull request by kontura**
For: https://github.com/rpm-software-management/dnf5/issues/1025

Test update: https://github.com/rpm-software-management/ci-dnf-stack/pull/1465

### CODE REVIEW

File: libdnf5/base/base_impl.hpp [hunk 1]
```diff
@@ -77,6 +80,18 @@ class Base::Impl {
...
+    repo::RepoSack repo_sack;
+    rpm::PackageSack rpm_package_sack;
+    module::ModuleSack module_sack;
+    std::map<std::string, std::string> variables;
+    transaction::TransactionHistory transaction_history;
+    Vars vars;
+    std::unique_ptr<repo::DownloadCallbacks> download_callbacks;
+
+    WeakPtrGuard<LogRouter, false> log_router_gurad;
+    WeakPtrGuard<Vars, false> vars_gurad;
```
Comment (jan-kolarik) [inline libdnf5/base/base_impl.hpp:94]: I know it was already there, but when we're already touching this part of code, please fix these typos:
- log_router_gurad -> log_router_guard
- vars_gurad -> vars_guard

**Review by jan-kolarik — APPROVED** [review]
Just the two typos, otherwise LGTM.

**Comment by jan-kolarik** [comment]
Failing tests are fixed in the enclosed CI PR.

File: libdnf5/base/base_impl.hpp (hunk 1 above)
Comment (kontura) [reply libdnf5/base/base_impl.hpp]: Nice catch, fixed.

**Pull request merged**